Analysis/HistProducerFile.py

```diff
@@ -0,0 +1,183 @@
+import ROOT
+import sys
+import os
+import math
+import shutil
+from RunKit.sh_tools import sh_call
+if __name__ == "__main__":
+    sys.path.append(os.environ['ANALYSIS_PATH'])
+from datetime import datetime
+
+import Common.Utilities as Utilities
+from Analysis.HistHelper import *
+from Analysis.RegionDefinition import *
+scales = ['Up', 'Down']
+'''
+def getHistValues(histograms):
+    for var in histograms.keys():
+        histograms_var = histograms[var]
+        for key_tuple in histograms_var:
+            final_hist = histograms_var[key_tuple][0].GetValue()
+            objsToMerge = ROOT.TList()
+            for hist in histograms_var[key_tuple][1:]:
+                objsToMerge.Add(hist.GetValue())
+                final_hist.Merge(objsToMerge)
+            histograms_var[key_tuple] = final_hist
+'''
+
+def createCentralQuantities(df_central, central_col_types, central_columns):
+    tuple_maker = ROOT.analysis.MapCreator(*central_col_types)()
+    tuple_maker.processCentral(ROOT.RDF.AsRNode(df_central), Utilities.ListToVector(central_columns))
+    tuple_maker.getEventIdxFromShifted(ROOT.RDF.AsRNode(df_central))
+
+def SaveHists(histograms, out_file):
+    for var in histograms.keys():
+        histograms_var = histograms[var]
+        for key_tuple,hist_list in histograms_var.items():
+            dir_ptr = make_dir(out_file, key_tuple[0])
+            merged_hist = hist_list[0].GetValue()
+            for hist in hist_list[1:] :
+                merged_hist.Add(hist.GetValue())
+            hist_name = '_'.join(key_tuple[1])
+            print(f"histName is {hist_name}")
+            dir_ptr.WriteObject(merged_hist, hist_name)
```

Comment:
Different vars go in different files. there should be no look over `histograms.keys()` here, it is already done below in L173.